src/toil/lib/toilDocker.py

```diff
@@ -0,0 +1,213 @@
+        raise ImageNotFound("Docker image not found.")
+    except APIError:
+        _logger.error("The server returned an error.")
+        raise APIError("The server returned an error.")
+    except:
+        raise ("An unexpected error occurred while attempting to call docker.")
+
+def _dockerKill(container_name, client):
+    """
+    Immediately kills a container.  Equivalent to "docker kill":
+    https://docs.docker.com/engine/reference/commandline/kill/
+
+    :param container_name: Name of the container being killed.
+    :param client: The docker API client object to call.
+    """
+    try:
+        this_container = client.containers.get(container_name)
+        if this_container.status == 'running':
+            this_container.kill(signal=None)
+            while (this_container.status == 'running'):
+                this_container = client.containers.get(container_name)
+    except NotFound:
+        _logger.debug("Attempted to kill container, but container does not exist: ", container_name)
+
+def _dockerStop(container_name, client):
+    """
+    Gracefully kills a container.  Equivalent to "docker stop":
+    https://docs.docker.com/engine/reference/commandline/stop/
+
+    :param container_name: Name of the container being stopped.
+    :param client: The docker API client object to call.
+    """
+    try:
+        this_container = client.containers.get(container_name)
+        while this_container.status == 'running':
+            try:
+                client.containers.get(container_name).stop()
+            except:
+                pass
+            this_container = client.containers.get(container_name)
+    except NotFound:
+        _logger.debug("Attempted to stop container, but container does not exist: ", container_name)
+
+def _containerIsRunning(container_name):
+    """
+    Checks whether the container is running or not.
+
+    :param container_name: Name of the container being checked.
+    :returns: True if status is 'running', False if status is anything else,
+    and None if the container does not exist.
+    """
+    client = docker.from_env()
+    try:
+        this_container = client.containers.get(container_name)
+        if this_container.status == 'running':
+            return True
+        elif this_container.status == 'exited':
+            return False
+        elif this_container.status == 'restarting':
+            return False
+        elif this_container.status == 'paused':
+            return False
+    except NotFound:
+        return None
+    except APIError:
+        _logger.debug("Server error attempting to call container: ", container_name)
+        raise
+
+def _getContainerName(job):
+    """Create a random string including the job name, and return it."""
+    return '--'.join([str(job), base64.b64encode(os.urandom(9), '-_')]).replace("'", '').replace('_', '')
+
+def getDeprecatedCmdlineParams(cmdline_params):
+    '''Not sure whether to include this yet.'''
```

Comment:
So are you going to include this?

Reply:
I've taken it out for now, but I will write one tonight.

Reply:
Updated.